File: code/cpp/medium/1353-maxEvents.cpp
class Solution {
    public:
     int maxEvents(vector<vector<int>>& events) {
         if (events.empty()) {
             return 0;
         }
 
         sort(events.begin(), events.end());
 
         priority_queue<int, vector<int>, greater<int>> min_heap;
         int max_events = 0;
         int i = 0;
         int n = events.size();
         int current_day = 0;
 
         while (!min_heap.empty() || i < n) {
             if (min_heap.empty()) {
                 current_day = events[i][0];
             }
 
             while (i < n && events[i][0] <= current_day) {
                 min_heap.push(events[i][1]);
                 i++;
             }
 
             min_heap.pop();
             max_events++;
             current_day++;
 
             while (!min_heap.empty() && min_heap.top() < current_day) {
                 min_heap.pop();
             }
         }
 
         return max_events;
     }
 };

File: code/java/medium/1353-maxEvents.java
class Solution {
    public int maxEvents(int[][] events) {
        if (events == null || events.length == 0) {
            return 0;
        }

        Arrays.sort(events, (a, b) -> Integer.compare(a[0], b[0]));

        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        int maxEvents = 0;
        int i = 0;
        int n = events.length;
        int currentDay = 0;

        while (!minHeap.isEmpty() || i < n) {
            if (minHeap.isEmpty()) {
                currentDay = events[i][0];
            }

            while (i < n && events[i][0] <= currentDay) {
                minHeap.offer(events[i][1]);
                i++;
            }

            minHeap.poll();
            maxEvents++;
            currentDay++;

            while (!minHeap.isEmpty() && minHeap.peek() < currentDay) {
                minHeap.poll();
            }
        }

        return maxEvents;
    }
}